### ISSUE TITLE
[FEATURE] Dependencies between Tasks

### ISSUE THREAD

**Issue by datWeazel** [Leantime/leantime#1103]
**Is your feature request related to a problem? Please describe.**
Let's say we have Task A, B and C. B and C can only be done once A is done. Currently there's no way to define dependencies between the tasks. 

There are sub tasks but those can't carry the same information as a regular task (like a description, file, or anything besides a title really)

**Describe the solution you'd like**
A solution similar to how [Asana handles dependencies](https://asana.com/guide/help/tasks/dependencies) would be nice.

**Comment by JRNexic** [comment]
My team would be thrilled to have this added.

**Comment by marcelfolaron** [comment]
I agree with this feature. 
My current thoughts are around creating a flexible "relationship" structure that allows dependencies between tasks but also relationships between ideas, milestones, canvas items, tasks etc. Jira handles it quite nicely by allowing users to set the "relationship" to depends on, blocked by,  is related to, duplicate of, etc. 

I am not sure when exactly I'll be able to get to it yet so if there is someone from the community interesting in starting this, this would be great.

**Comment by salmonix** [comment]
A simple directed dependency graph would be very useful and in my impression enough where the milestones are single nodes or 'barriers'. 
What really appeals me in Leantime is the simplicity that is easy to lose. Jira etc. are - in my impression - for complex organizations and usually messy structure/backlog/team domains where hierarchies are built via Scrum etc., and that is expressed in the complexity of the tools. For teams where the teams own project management is in focus and hierarchies are flat - like in real Agile as per Agile manifesto - this complexity is unnecessary. I was lucky to experience real Agile work for a few years - and we did not need the 99% of the Jira features/tags etc., but missed this graph visualization, for instance.
